backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxSparkPlanExecApi.scala

```diff
@@ -181,6 +181,11 @@ class VeloxSparkPlanExecApi extends SparkPlanExecApi {
       function: ExpressionTransformer,
       expr: ArrayFilter): ExpressionTransformer = {
     expr.function match {
+      // Transformer for array_compact.
+      case LambdaFunction(function, arguments, _)
+          if function.getClass.getSimpleName.equals("IsNotNull") &&
+            function.children == arguments =>
```

Comment:
Is this check for arguments necessary?

Reply:
You are right, it's unnecessary under ArrayFilter expression. Removed